core/src/test/java/org/everit/json/schema/internal/DefaultFormatValidatorTest.java

```diff
@@ -44,18 +44,64 @@ private void assertSuccess(final String subject, final FormatValidator format) {
     @Test
     public void dateTimeExceedingLimits() {
         assertFailure("1996-60-999T16:39:57-08:00", new DateTimeFormatValidator(),
-                "[1996-60-999T16:39:57-08:00] is not a valid date-time");
+                "[1996-60-999T16:39:57-08:00] is not a valid date-time. Expected [yyyy-MM-dd'T'HH:mm:ssZ, yyyy-MM-dd'T'HH:mm:ss.[0-9]{1,9}Z]");
     }
 
     @Test
     public void dateTimeFormatFailure() {
         assertFailure("2015-03-13T11:00:000", new DateTimeFormatValidator(),
-                "[2015-03-13T11:00:000] is not a valid date-time");
+                "[2015-03-13T11:00:000] is not a valid date-time. Expected [yyyy-MM-dd'T'HH:mm:ssZ, yyyy-MM-dd'T'HH:mm:ss.[0-9]{1,9}Z]");
+    }
+
+    @Test
+    public void dateTimeWithSingleDigitInSecFracSuccess() {
+        assertSuccess("2015-02-28T11:00:00.1Z", new DateTimeFormatValidator());
+    }
+
+    @Test
+    public void dateTimeWithTwoDigitsInSecFracSuccess() {
+        assertSuccess("2015-02-28T11:00:00.12Z", new DateTimeFormatValidator());
+    }
+
+    @Test
+    public void dateTimeWithThreeDigitsInSecFracSuccess() {
+        assertSuccess("2015-02-28T11:00:00.123Z", new DateTimeFormatValidator());
+    }
+
+    @Test
+    public void dateTimeWithFourDigitsInSecFracSuccess() {
+        assertSuccess("2015-02-28T11:00:00.1234Z", new DateTimeFormatValidator());
+    }
+
+    @Test
+    public void dateTimeWithFiveDigitsInSecFracSuccess() {
+        assertSuccess("2015-02-28T11:00:00.12345Z", new DateTimeFormatValidator());
+    }
+
+    @Test
+    public void dateTimeWithSixDigitsInSecFracSuccess() {
+        assertSuccess("2015-02-28T11:00:00.123456Z", new DateTimeFormatValidator());
+    }
+
+    @Test
+    public void dateTimeWithSevenDigitsInSecFracSuccess() {
+        assertSuccess("2015-02-28T11:00:00.1234567Z", new DateTimeFormatValidator());
+    }
+
+    @Test
+    public void dateTimeWithEightDigitsInSecFracSuccess() {
+        assertSuccess("2015-02-28T11:00:00.12345678Z", new DateTimeFormatValidator());
+    }
+
+    @Test
+    public void dateTimeWithNineDigitsInSecFracSuccess() {
+        assertSuccess("2015-02-28T11:00:00.123456789Z", new DateTimeFormatValidator());
     }
 
     @Test
-    public void dateTimeSecFracSuccess() {
-        assertSuccess("2015-02-28T11:00:00.111Z", new DateTimeFormatValidator());
```

Comment:
The [Json Schema Validation spec](http://json-schema.org/latest/json-schema-validation.html#rfc.section.7.3.1) seem to suggest that the date-time format need to specifically conform to [Section 5.6 of RFC-3339](https://tools.ietf.org/html/rfc3339#section-5.6). 
I think Section 5.6 of RFC-3339 is a specific cut down version of the ISO 8601 grammar and the [ABNF date time](https://tools.ietf.org/html/rfc3339#appendix-A) that your referring to in the above comment, is an ISO 8601 standard.
Hope it helps.